newton/_src/sim/joints.py

```diff
@@ -18,17 +18,52 @@
 
 # Types of joints linking rigid bodies
 class JointType(IntEnum):
+    """
+    Enumeration of joint types supported in Newton.
+    """
+
     PRISMATIC = 0
+    """Prismatic joint: allows translation along a single axis (1 DoF)."""
+
     REVOLUTE = 1
+    """Revolute joint: allows rotation about a single axis (1 DoF)."""
+
     BALL = 2
+    """Ball joint: allows rotation about all three axes (3 DoF, quaternion parameterization)."""
+
     FIXED = 3
+    """Fixed joint: locks all relative motion (0 DoF)."""
+
     FREE = 4
+    """Free joint: allows full 6-DoF motion (translation and rotation, 7 coordinates)."""
+
     DISTANCE = 5
+    """Distance constraint: maintains a fixed distance between two points (6 DoF, 7 coordinates)."""
+
     D6 = 6
+    """6-DoF joint: allows translation and rotation about all axes, with individual axis locking."""
```

Comment:
Something along this description from https://newton-physics.github.io/newton/concepts/articulations.html might be a bit clearer:
> Generic D6 joint with up to 3 translational and 3 rotational degrees of freedom